src/main/java/com/hibernate/cinema/controller/MovieSessionController.java

```diff
@@ -0,0 +1,46 @@
+package com.hibernate.cinema.controller;
+
+import com.hibernate.cinema.model.dto.MovieSessionRequestDto;
+import com.hibernate.cinema.model.dto.MovieSessionResponseDto;
+import com.hibernate.cinema.service.MovieSessionService;
+import com.hibernate.cinema.service.mapper.MovieSessionMapper;
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.util.List;
+import java.util.stream.Collectors;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequestMapping("/movie-sessions")
+public class MovieSessionController {
+    private static final String DATE_FORMAT = "d-MM-yyyy";
+    private final MovieSessionService movieSessionService;
+    private final MovieSessionMapper movieSessionMapper;
+
+    @Autowired
+    public MovieSessionController(MovieSessionService movieSessionService,
+                                  MovieSessionMapper movieSessionMapper) {
+        this.movieSessionService = movieSessionService;
+        this.movieSessionMapper = movieSessionMapper;
+    }
+
+    @PostMapping
+    public void addMovieSession(@RequestBody MovieSessionRequestDto movieSessionDto) {
+        movieSessionService.add(movieSessionMapper.castDtoToMovieSession(movieSessionDto));
+    }
+
+    @GetMapping("/available")
+    public List<MovieSessionResponseDto> getAvailable(@RequestParam Long movieId,
+                                                      @RequestParam String date) {
```

Comment:
this is optional, but you can consider following approach 
@RequestParam @DateTimeFormat(pattern = "dd.MM.yyyy") LocalDate date